jakarta.ls/src/main/resources/servlet.json

```diff
@@ -76,5 +76,33 @@
         "context":{
             "type": "jakarta.servlet.http.HttpServlet"
         }
-    }
+    },
+    "Jakarta Servlet WebFilter": {
+        "prefix": "servletfilter",
+        "body": [
+          "package ${1:packagename};",
+          "",
+          "/* Filter Dependencies */",
+          "import jakarta.servlet.Filter;",
+          "import jakarta.servlet.FilterChain;",
+          "import jakarta.servlet.ServletException;",
+          "import jakarta.servlet.ServletRequest;",
+          "import jakarta.servlet.ServletResponse;",
+          "import jakarta.servlet.annotation.WebFilter;",
+          "",
+          "import java.io.IOException;",
+          "",
+          "@WebFilter(filterName = \"${2:filterName}\", urlPatterns = {\"${3:urlPattern}\"})",
+          "public class ${4:className} implements Filter {",
```

Comment:
For the class name you could try using `${TM_FILENAME_BASE}`. Ex. https://github.com/eclipse/lsp4mp/blob/master/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/resources/org/eclipse/lsp4mp/snippets/mp-health.json#L15

I am not sure if it works in eclipse so u would have to give it a try.

Reply:
If this works we should update the other servlet snippets as well

Reply:
I just did some testing, and the proposed method doesnt actually work in eclipse rn. I tried messing around with templates in eclipse and that does fill in snippets with variables as we want but for whatever reason that is not extending to our current completion framework. So for now I would say we should keep snippets like this. In order to accomodate the change in this request I think i would have to restructure completionProposalComputer a bit and I am not sure about what is needed for that yet

Reply:
Sounds good, thanks for opening this issue #43 Ankush, we can leave the snippets as-is for now and address this at a later date.